### PR TITLE
[2.11.x] DDF-3619 updated map layers configuration docs

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Draft of documentation for configuring map layers in UI. 

#### Who is reviewing it? 

@emmberk @mcalcote @vinamartin @ahoffer @garrettfreibott @djblue 

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@lessarderic
@ricklarsen - Documentation
@rzwiefel
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

review content for accuracy. 

#### What are the relevant tickets?
[DDF-3619](https://codice.atlassian.net/browse/DDF-3619)

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_configuring/map-layers-intrigue.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+.Deleting a Map Layer
+. Delete an unneeded map layer with the *Delete Layer*(image:trash.png[trash icon]) icon associated with that layer.
+
+To remove all map layers, select *RESET*.
+
+.Reordering Map Layers
+. Move layers *Up* and *Down* in loading order with the *Arrow Icons* associated with each layer.
+
+
```
Comment (emmberk) [inline distribution/docs/src/main/resources/content/_configuring/map-layers-intrigue.adoc]: Extra line

File: distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Customize the look of the map displayed to users in ${ddf-ui} by adding or removing map layers through the Map Layers UI.
+
+. Navigate to the *${admin-console}*.
+. Select the *${ddf-ui}* application.
+. Select the *Configuration* tab.
+. Select the *Standard Search UI* configuration.
+
+.Adding a Map Layer
+. Select the *add* button to add a new layer
```
Comment (emmberk) [inline distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc]: Missing a period at the end of this sentence

File: distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc [hunk 3]
```diff
@@ -8,39 +8,51 @@
...
 |Format
-|Status
+|2D Documentation
+|3D Documentation
 
-|https://cesiumjs.org/Cesium/Build/Documentation/createOpenStreetMapImageryProvider.html[Open Street Map]
-|Supported
+|Open Street Map
+|https://openlayers.org/en/latest/apidoc/ol.source.OSM.html[Open Layers]
+|https://cesiumjs.org/Cesium/Build/Documentation/createOpenStreetMapImageryProvider.html[Cesium]
```
Comment (emmberk) [inline distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: I think these links OpenLayers and Cesium links should point to the documentation for the current version of cesium that is used in DDF, not just the documentation for the latest version.

File: distribution/docs/src/main/resources/content/_appReferences/mg-geowebcache.adoc [hunk 4]
```diff
@@ -137,7 +135,7 @@ Add a new Imagery Provider in the ${admin-console}:
 . Select *Standard Search UI* configuration.
 . Click the *Add* button next to *Imagery Providers*
 . Enter configuration for Imagery Provider in new textbox:
-. `{"type" "WMS" "url" "${secure_url}/geowebcache/service/wms" "layers" ["states"] "parameters" {"FORMAT" "image/png"} "alpha" 0.5}`
+. `{"type" "WMS" "url" "${secure_url}/geowebcache/service/wms" "layers" ["states"] "parameters" {"FORMAT" "image/png" "VERSION" "1.1.1"} "alpha" 0.5}`
```
Comment (emmberk) [inline distribution/docs/src/main/resources/content/_appReferences/mg-geowebcache.adoc]: ❗ There are also examples in https://github.com/codice/ddf/blob/2c280fca47752d4f0fe573827e815def461fd02d/distribution/docs/src/main/resources/content/_tables/search.standard.properties.adoc and https://github.com/codice/ddf/blob/12eb736abc9609426d88af7fa15a445ca0b44c77/distribution/docs/src/main/resources/content/_tables/ui.config%202.adoc that need to be update.

File: distribution/docs/src/main/resources/content/_configuring/map-layers-intrigue.adoc (hunk 1 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_configuring/map-layers-intrigue.adoc]: fixed.

File: distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc (hunk 2 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc]: fixed.

File: distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc (hunk 3 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: I agree.

Comment (emmberk) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: Looks like catalog-ui-search uses openlayers 3.20.1 https://github.com/codice/ddf/blob/4e27c8f96b0119dceab9de21afe6dc6df24f9571/catalog/ui/catalog-ui-search/package.json#L96 (or maybe 3.16.0? https://github.com/codice/ddf/blob/e8163ed0493711cd2724e1657e14046f95209c21/catalog/ui/search-ui/standard/pom.xml#L266-L270) and cesium 1.34 https://github.com/codice/ddf/blob/4e27c8f96b0119dceab9de21afe6dc6df24f9571/catalog/ui/catalog-ui-search/package.json#L79  (or maybe 1.22.0? https://github.com/codice/ddf/blob/e8163ed0493711cd2724e1657e14046f95209c21/catalog/ui/search-ui/standard/pom.xml#L151-L155).

So the links in this file and https://github.com/codice/ddf/blob/a45417ec2878dc76dcf5dfaf621887fb5ead3622/catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/options.json should be updated to point to these specific versions. I'm having trouble finding a direct link to the javadoc for these not-latest versions. I think updating these to say `See the OpenLayers 3.20.1 documentation at http://openlayers.org/` and `See the Cesium 1.34 documentation at https://cesiumjs.org` would be better than linking directly to the latest docs.

Comment (emmberk) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: Also, I think OpenLayers might be one word: http://openlayers.org/en/latest/doc/tutorials/introduction.html.

Comment (brendan-hofmann) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: Shouldn't those pom and package.json versions match up, and then you can just property it into the docs as well?

Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: That's what I'm hoping to do

Comment (mackncheesiest) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: I think the package.json version should be the one used by yarn/npm/whatever we use to actually fetch the dependencies on install. Not sure what the webjar dependencies are necessarily for, though.

Comment (djblue) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: I would also go with the package.json version

Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: Should the others be a) made to match? b) deleted entirely? c) left alone?

Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_coreConcepts/map-standards.adoc]: looking at these again, it appears the difference is one version is intrigue and the other is the standard search ui.

**Comment by ricklarsen** [comment]
updated per changes requested by @emmberk & @djblue

File: distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+. Select the *add* button to add a new layer.
+. Enter the properties of the map layer into the *Imagery Provider* in the proper syntax.
+.. Example Imagery Provider Syntax: `{"type" "WMS" "url" "http://example.com" "layers" ["layer1" "layer2"] "parameters" {"FORMAT" "image/png" "VERSION" "1.1.1"} "alpha" 0.5}`.
+... "type": format of imagery provider.
+... "url": location of server hosting the imagery provider.
+... "layers": names of individual layers. (enclose list in square brackets`[ ]`).
+... "parameters": (enclose in braces `{}`):
+.... "FORMAT": image type used by imagery provider.
+.... "VERSION": Version of imagery provider to use.
+.... "alpha": opacity of imagery provider layer.
```
Comment (emmberk) [inline distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc]: Nit: update to consistent capitalization after the colon here:
```
.... "FORMAT": image type used by imagery provider.
.... "VERSION": version of imagery provider to use.
.... "alpha": opacity of imagery provider layer.
```

Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_configuring/map-layers-standard.adoc]: fixed

**Comment by ricklarsen** [comment]
build now

File: catalog/ui/search-ui/standard/src/main/resources/OSGI-INF/metatype/metatype.xml [hunk 6]
```diff
@@ -23,14 +23,14 @@
         </AD>
         <AD name="Imagery Providers" id="imageryProviders"
             required="false" type="String" cardinality="10000"
-            description='List of imagery providers to use. Valid types are: OSM (OpenStreetMap), AGM (ArcGisMap), BM (BingMap), WMS (WebMapService), WMT (WebMapTile), TMS (TileMapService), GE (GoogleEarth). Example: {"type" "WMS" "url" "http://example.com" "layers" ["layer1" "layer2"] "parameters" {"FORMAT" "image/png" "VERSION" "1.1.1"} "alpha" 0.5}'
+            description='List of imagery providers to use. Valid types are: OSM (OpenStreetMap), AGM (ArcGisMap), BM (BingMap), WMS (WebMapService), WMT (WebMapTile), TMS (TileMapService), GE (GoogleEarth). Example: {"name": "Example OSM", "show": true, "type": "OSM", "url": "http://a.tile.openstreetmap.org", "fileExtension": "png",  "order": 0, "alpha": 1.0 }'
```
Comment (bdeining) [inline catalog/ui/search-ui/standard/src/main/resources/OSGI-INF/metatype/metatype.xml]: I know that the standard search ui was using a different formatting style (possibly the one before the changes here).  Has this configuration been verified?

File: distribution/docs/src/main/resources/content/_appReferences/mg-geowebcache.adoc [hunk 7]
```diff
@@ -137,12 +135,12 @@ Add a new Imagery Provider in the ${admin-console}:
 . Select *Standard Search UI* configuration.
 . Click the *Add* button next to *Imagery Providers*
 . Enter configuration for Imagery Provider in new textbox:
-. `{"type" "WMS" "url" "${secure_url}/geowebcache/service/wms" "layers" ["states"] "parameters" {"FORMAT" "image/png"} "alpha" 0.5}`
+. `{"type": "WMS", "order": 0, "url": "${secure_url}/geowebcache/service/wms", "layers": ["states"], "parameters": {"FORMAT": "image/png", "VERSION": "1.1.1"}, "alpha" 0.5}`
```
Comment (bdeining) [inline distribution/docs/src/main/resources/content/_appReferences/mg-geowebcache.adoc]: Same question about the Standard Search UI format.  If they are different we may want to provide both configuration styles in this documentation.

File: distribution/docs/src/main/resources/content/_configuring/map-layers-intrigue.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+== {title}
+
+Customize the look of the map displayed to users in ${catalog-ui} by adding or removing map layers through the *Map Layers UI*.
+
+. Navigate to the *${admin-console}*.
+. Select the *${ddf-catalog}* application.
+. Select the *Map Layers* tab.
+. *Add*, *Configure* or *Remove* map layers as desired.
+
+.Adding a Map Layer
```
Comment (bdeining) [inline distribution/docs/src/main/resources/content/_configuring/map-layers-intrigue.adoc]: We may want to point out that different map layers require different parameters.  For example `WMTS` may require a set of `tileMatrixLabels`.  See https://github.com/codice/ddf/pull/1763 for an example.

File: catalog/ui/search-ui/standard/src/main/resources/OSGI-INF/metatype/metatype.xml (hunk 6 above)
Comment (ricklarsen) [reply catalog/ui/search-ui/standard/src/main/resources/OSGI-INF/metatype/metatype.xml]: reverting

File: distribution/docs/src/main/resources/content/_appReferences/mg-geowebcache.adoc (hunk 7 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_appReferences/mg-geowebcache.adoc]: reverting

**Comment by ricklarsen** [comment]
build now

**Comment by ricklarsen** [comment]
rebased to rerun 
build now

**Comment by ricklarsen** [comment]
build now

**Comment by ricklarsen** [comment]
yarn failure, rerunning
build now

**Comment by ricklarsen** [comment]
Cancelled job to restart. build now

**Comment by clockard** [comment]
build failed for owasp download failure. Marking the build as passed since there are no dependency updates.

**Pull request merged**